rscripts/YAL5-single-codons.R

```diff
@@ -0,0 +1,1108 @@
+    geom_line() +
+    theme_bw() +
+    theme(text=element_text(size=14),
+          axis.title=element_text(size=14, face="bold"),
+          title = element_text(size = 14, face="bold"))+
+    labs(title = paste0("Relative read counts around feature ", feature_of_interest),
+         x = "Position relative to feature of interest",
+         y = "Relative read count", size = 2) + 
+    scale_x_continuous(breaks = seq(-expand_width, expand_width, 2))
+  
+  # save plot as PDF
+  
+  print("Save plot as PDF")
+  save_plot_pdf <- function(overlayed_plot, output_dir){
+    overlayed_plot %>%
+      ggsave(
+        filename = file.path(output_dir,
+                             paste0("Meta_feature_plot_", 
+                                    feature_of_interest, "_", dataset,".pdf")),
+        width = 6, height = 5
+      )
+  }
+  
+  save_plot_pdf(overlayed_plot, output_dir)
+  
+  print("Done")
+  
+  
+} else{
+  
+  # FindAllFeatures is a function that contains ExpandFeatureRegionAllGenes,
+  # ExpandedRegionNormalization and OverlayedTable, which are defined above.
+  # The function creates a tibble of features and their relcounts at position 
+  # 0 of their overlayed tibble which can e used to compare the relcount between different features.
+  
+  FindAllFeatures <- function(gene_names, dataset, hd_file, 
+                              min_read_length,   
+                              gff_df, filter_for_frame,
+                              filtering_frame, snapdisp, yeast_codon_pos_i200,
+                              .x,
+                              expand_width, 
+                              remove_overhang
+  ){
+    
+    # set .x to feature being studies, allowing purrr::map to iterate over 
+    # different features of interest without having problems due to .x being a changing vector
+    
+    feature_being_studied <- .x
+    
+    
+    # Run ExpandFeatureRegionAllGenes to get a list of occurrances of the codon of interest 
+    
+    print(paste0("Finding occurances of ", feature_being_studied))
+    output_feature_info <- suppressMessages(
+      ExpandFeatureRegionAllGenes(gene_names, dataset, hd_file, 
+                                  min_read_length,   
+                                  gff_df, filter_for_frame,
+                                  filtering_frame, snapdisp, yeast_codon_pos_i200,
+                                  feature_of_interest = feature_being_studied,
+                                  expand_width, 
+                                  remove_overhang
+      ))
+    
+    
+    # Check for the presence of the feature of interest. 
+    # Output_feature_info being empty will cause problems with normalization
+    
+    if(length(output_feature_info) == 0){
+      print(paste("No occurrances of", feature_being_studied))
+      
+      if(expand_width > 1){
+        
+        print("Use an expand_width of 1L to check for occurances near to start or stop codon")
```

Comment:
occurances => occurrences
near to start or stop codon => near start or stop codons